mlir/include/mlir/Dialect/MIOpen/Pipeline.h

```diff
@@ -0,0 +1,65 @@
+//===- Pipeline.h - MIOpen pipeline creation ---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the miopen::addPipeline routine to consolidate the
+// MIOpen pipeline creation.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_MIOPEN_PIPELINE_H
+#define MLIR_DIALECT_MIOPEN_PIPELINE_H
+
+#include <mlir/Pass/PassManager.h>
+
+#include <string>
+
+namespace mlir {
+namespace miopen {
+
+struct PipelineConfig {
+  bool highLevelInput = false;
+  bool tuningTest = false;
+  bool runBackend = false;
+  struct PerfConfig {
+    std::string config;
+    int64_t gridSize = 0;
+    int64_t blockSize = 0;
+  } perf;
+  struct TargetConfig {
+    std::string triple;
+    std::string chip;
+    std::string features;
+  } target;
+};
+
+struct TuningPipeline : PipelineConfig {
+  TuningPipeline(const std::string &perfConfig, int64_t gridSize = 0,
+                 int64_t blockSize = 0)
+      : PipelineConfig{false, true, false, {perfConfig, gridSize, blockSize}} {}
+};
+
+struct DriverPipeline : PipelineConfig {
+  DriverPipeline(const std::string &perfConfig)
+      : PipelineConfig{false, false, false, {perfConfig}} {}
+};
+
+template <bool highLevel = false> struct KernelPipeline : PipelineConfig {
```

Comment:
Why does a pipeline inherits from the config? I feel like the relationship should be composition, not inheritance.

Reply:
It is the kind of PipelineConfig, see usage. They must derive from PipelineConfig to be passed into addPipeline.

Reply:
Okay then could you rename it as `KernelPipelineConfig`.

Also, why use the template with the default bool? If you are using this as a thin wrapper you can expose it in the constructor.